ckanext/xloader/action.py

```diff
@@ -40,6 +40,10 @@ def xloader_submit(context, data_dict):
     :param ignore_hash: If set to True, the xloader will reload the file
         even if it haven't changed. (optional, default: False)
     :type ignore_hash: bool
+    :param sync: If set to True, the xloader callback will be executed right
```

Comment:
I'm not comfortable with giving this power to all publishers. It would become incredibly easy to DoS the site by accident. Remember that synchronous processing will use the memory and CPU cycles of the web server.

It's different from providing a flag on the command line. That can only be used by staff who have top-level access and should be familiar with the technical capabilities of the system. This, though, is potentially open to non-technical users who only know that they're being pressured to make things happen faster.

If this feature is going to be present, it ought to require higher level authorisation such as a sysadmin.

Reply:
OMG I forgot that `xloader_submit` is a registered action. Yeah that is bad haha. Okay I will move the sync stuff around and see if I can figure out a bit more